src/strategies/dutchv3_strategy.rs

```diff
@@ -0,0 +1,389 @@
+    utils::hex,
+};
+use std::error::Error;
+use std::str::FromStr;
+use std::sync::Arc;
+use std::{collections::{HashMap, HashSet}, fmt::Debug};
+use tokio::sync::mpsc::{Receiver, Sender};
+use tracing::{error, info};
+use uniswapx_rs::order::{Order, OrderResolution, V3DutchOrder};
+
+use super::types::{Action, Event};
+
+const DONE_EXPIRY: u64 = 300;
+const REACTOR_ADDRESS: &str = "0xB274d5F4b833b61B340b654d600A864fB604a87c";
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct UniswapXDutchV3Fill<M> {
+    /// Ethers client.
+    client: Arc<M>,
+    /// executor address
+    executor_address: String,
+    /// Amount of profits to bid in gas
+    bid_percentage: u64,
+    last_block_number: u64,
+    last_block_timestamp: u64,
+    // map of open order hashes to order data
+    open_orders: HashMap<String, OrderData>,
+    // map of order hashes that are currently being processed (routed/executed)
+    processing_orders: HashSet<String>,
+    // map of done order hashes to time at which we can safely prune them
+    done_orders: HashMap<String, u64>,
+    batch_sender: Sender<Vec<OrderBatchData>>,
+    route_receiver: Receiver<RoutedOrder>,
+}
+
+impl<M: Middleware + 'static> UniswapXDutchV3Fill<M> {
+    pub fn new(
+        client: Arc<M>,
+        config: Config,
+        sender: Sender<Vec<OrderBatchData>>,
+        receiver: Receiver<RoutedOrder>,
+    ) -> Self {
+        info!("syncing state");
+
+        Self {
+            client,
+            executor_address: config.executor_address,
+            bid_percentage: config.bid_percentage,
+            last_block_number: 0,
+            last_block_timestamp: 0,
+            open_orders: HashMap::new(),
+            processing_orders: HashSet::new(),
+            done_orders: HashMap::new(),
+            batch_sender: sender,
+            route_receiver: receiver,
+        }
+    }
+}
+
+#[async_trait]
+impl<M: Middleware + 'static> Strategy<Event, Action> for UniswapXDutchV3Fill<M> {
+    // In order to sync this strategy, we need to get the current bid for all Sudo pools.
+    async fn sync_state(&mut self) -> Result<()> {
+        info!("syncing state");
+
+        Ok(())
+    }
+
+    // Process incoming events, seeing if we can arb new orders, and updating the internal state on new blocks.
+    async fn process_event(&mut self, event: Event) -> Vec<Action> {
+        match event {
+            Event::UniswapXOrder(order) => self.process_order_event(&order).await
+                .map_or(vec![], |action| vec![action]),
+            Event::NewBlock(block) => self.process_new_block_event(&block).await
+                .map_or(vec![], |action| vec![action]),
+            Event::UniswapXRoute(route) => self.process_new_route(&route).await
+                .map_or(vec![], |action| vec![action]),
+        }
+    }
+}
+
+impl<M: Middleware + 'static> UniswapXStrategy<M> for UniswapXDutchV3Fill<M> {}
+
+struct DutchV3OrderWrapper {
+    inner: V3DutchOrder,
+    encoded_order: String,
```

Comment:
For some reason the re-encoding in Rust was slightly different than the original. I triple checked the struct and couldn't find any differences. To get around this, I just save the encoded_order and use it to send the actual tx.

Reply:
The abis / generated crate might be outdated, I would double check that by removing everything and generating again from scratch from within the contracts repo

Reply:
Yeah, I checked that as well.

Reply:
@ConjunctiveNormalForm do you have any ideas what could cause the encoding difference?